historydag/dag.py

```diff
@@ -384,15 +384,40 @@ def merge(self, other: "HistoryDag"):
         nodedict = {n: n for n in selforder}
         for n in otherorder:
             if n in nodedict:
-                pnode = nodedict[n]
+                pnode = nodedict[n]  # Get self's node in memory
             else:
                 pnode = n.node_self()
-                nodedict[n] = pnode
+                nodedict[n] = pnode  # Create and add node to dict
 
             for _, edgeset in n.clades.items():
                 for child, weight, _ in edgeset:
+                    # Add edge from parent to each child. Postorder
+                    #   traversal means new children will be in self
                     pnode.add_edge(nodedict[child], weight=weight)
 
+    def merge_list(self, treelist: List["HistoryDag"]):
```

Comment:
I would be fine with just replacing the `merge` method with this one, and adding a check with `isinstance` which, if the argument is a `HistoryDag` object, just wraps it in a list

Reply:
Ok I've made the change! One thing that's a little weird about having a merge method that accepts either `List[HistoryDag]` or `HistoryDag` (and then turns it into a list) is that it's not clear what the type of input object should be. I have the method header as `def merge(self, trees: List["HistoryDag"]):` and all the tests are passing (so I guess the type checker isn't checking that hard lol), but let me know if you'd like something different

Reply:
Awesome! Yeah I guess the type checker must not be running in the automatic tests at all haha. You can do `Union["HistoryDag", List["HistoryDag"]]` to express that the argument can have both types.

Reply:
Also don't forget to make it clear in the docstring that the method can accept either another HistoryDag, or a sequence of HistoryDags.